File: sdk/python/hitechcloud/resources/will_expired.py
"""WillExpired resource (10 endpoints)"""

from .base import BaseResource

class WillExpiredResource(BaseResource):
    """WillExpired: expiring services and domains management"""

    def list_expiring(self, **kwargs) -> dict:
        """GET /api/willexpired - List expiring services and domains"""
        return self._http.get("/api/willexpired", kwargs)

    def get_summary(self, **kwargs) -> dict:
        """GET /api/willexpired/summary - Expiring summary"""
        return self._http.get("/api/willexpired/summary", kwargs)

    def get_config(self) -> dict:
        """GET /api/willexpired/config - Module configuration"""
        return self._http.get("/api/willexpired/config")

    def list_open_invoices(self) -> dict:
        """GET /api/willexpired/invoices - Open renewal invoices"""
        return self._http.get("/api/willexpired/invoices")

    def list_requests(self) -> dict:
        """GET /api/willexpired/requests - Renewal request log"""
        return self._http.get("/api/willexpired/requests")

    def export_all(self) -> dict:
        """GET /api/willexpired/export - Export all expiring items"""
        return self._http.get("/api/willexpired/export")

    def get_item(self, item_type: str, item_id: int) -> dict:
        """GET /api/willexpired/{type}/{id} - Get item detail"""
        return self._http.get(f"/api/willexpired/{item_type}/{item_id}")

    def renew(self, item_type: str, item_id: int) -> dict:
        """POST /api/willexpired/{type}/{id}/renew - Renew item"""
        return self._http.post(f"/api/willexpired/{item_type}/{item_id}/renew")

    def get_autorenew(self, item_type: str, item_id: int) -> dict:
        """GET /api/willexpired/{type}/{id}/autorenew - Get auto-renew status"""
        return self._http.get(f"/api/willexpired/{item_type}/{item_id}/autorenew")

    def set_autorenew(self, item_type: str, item_id: int, enabled: bool) -> dict:
        """PUT /api/willexpired/{type}/{id}/autorenew - Set auto-renew"""
        return self._http.put(f"/api/willexpired/{item_type}/{item_id}/autorenew", {"enabled": enabled})

File: sdk/ruby/lib/hitechcloud/resources/will_expired.rb
module HiTechCloud
  # WillExpired resource
  class WillExpiredResource < BaseResource

    # List expiring items
    def list
      @http.get("/api/willexpired")
    end

    # Get summary
    def summary
      @http.get("/api/willexpired/summary")
    end

    # Get config
    def config
      @http.get("/api/willexpired/config")
    end

    # Get invoices
    def invoices
      @http.get("/api/willexpired/invoices")
    end

    # Get requests
    def requests
      @http.get("/api/willexpired/requests")
    end

    # Export
    def export
      @http.get("/api/willexpired/export")
    end

    # Get item
    def item(type, id)
      @http.get("/api/willexpired/%{type}/%{id}" % { type: type, id: id })
    end

    # Renew
    def renew(type, id, data)
      @http.post("/api/willexpired/%{type}/%{id}/renew" % { type: type, id: id }, data)
    end

    # Get autorenew
    def autorenew(type, id)
      @http.get("/api/willexpired/%{type}/%{id}/autorenew" % { type: type, id: id })
    end

    # Set autorenew
    def set_autorenew(type, id, data)
      @http.put("/api/willexpired/%{type}/%{id}/autorenew" % { type: type, id: id }, data)
    end

  end
end
